### PR TITLE
Simple edge discovery: spelling and linting

### PR DESCRIPTION
**Pull request by nicolacdnll**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:

Abide with the Megalinter workflow and spelling. **Changes to the API itself need a dedicated PR.**

Current linting is complaining about these two lines that are a regex and I'm not sure how to hard wrap. In any case, we can use equivalent regex that are also shorter. But that's for another PR. 

```yaml
- pattern: "^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$"
- pattern: "^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))$" 
```

image

#### Which issue(s) this PR fixes:

None

### CODE REVIEW

File: code/API_definitions/Discovery/README.md [hunk 1]
```diff
@@ -1,50 +1,60 @@
 # Edge Discovery APIs
 
 ## Simple Discovery API
-This API allows a client application to discover the closest MEC platform to the UE hosting the client application. 'Closest' means 'shorteset network path' as that will give the shortest propogation distance, which is a major factor in latency.
+
+This API allows a client application to discover the closest MEC platform to the UE hosting the client application. 'Closest' means 'shortest network path' as that will give the shortest propagation distance, which is a major factor in latency.
```
Comment (gunjald) [inline code/API_definitions/Discovery/README.md:5]: Should it be "platforms"? Plural as the API returns the array of platforms.

File: code/API_definitions/Discovery/README.md [hunk 2]
```diff
@@ -1,50 +1,60 @@
...
-1.	“I can retrieve a list of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)”
-2.	"I can discover the capabilities/resources available at an operator’s MEC: CPU, Memory, Storage, GPU"
-3.	"I can discover the geographical regions covered by the operators MECs"
-4.	"I can discover the closest MEC platform to a specific terminal (closest in terms of shortest network path)"
 
-16.	“I can ask the operator to provide the details of all the onboarded applications”
-17.	"I can ask the operator to inform about the application instance details e.g., communication endpoints, resource consumed etc"
+#### Provisioning intents
+
+1. "I can retrieve a list of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)"
```
Comment (gunjald) [inline code/API_definitions/Discovery/README.md:33]: We have terms with alternate words e.g. here it says MECs but the API says MEC Platforms

File: code/API_definitions/Discovery/README.md [hunk 3]
```diff
@@ -1,50 +1,60 @@
...
+16. "I can ask the operator to provide the details of all the on-boarded applications"
+17. "I can ask the operator to inform about the application instance details e.g., communication endpoints, resource consumed etc"
 
-#### Runtime intents 
-19.    "I can discover the closest MEC platform to a particular terminal (closest in terms of shortest network path)"
-20.    "I can discover the optimal MEC platform for my application and a particular terminal, taking into account connectivity, shortest network path, cost, network load etc." (`A`)
-21.    "I can discover the optimal application service endpoint for a specific terminal, taking into account mobility events, connectivity, shortest network path, cost, network load, MEC platform load etc."
+#### Runtime intents
+
+19. "I can discover the closest MEC platform to a particular terminal (closest in terms of shortest network path)"
```
Comment (gunjald) [inline code/API_definitions/Discovery/README.md:43]: Here we have MEC platform while at line 33 it is MECs

File: code/API_definitions/Discovery/simple_edge_discovery.yaml [hunk 4]
```diff
@@ -7,7 +7,7 @@ info:
     ---
     # Summary
 
-    The Simple Edge Discovery API returns the name of the closest operator MEC platform to a particular user device. 
+    The Simple Edge Discovery API returns the name of the closest operator MEC platform to a particular user device.
```
Comment (gunjald) [inline code/API_definitions/Discovery/simple_edge_discovery.yaml:10]: Should it be "platforms"? Singular to Plural?

Comment (nicolacdnll) [reply code/API_definitions/Discovery/simple_edge_discovery.yaml]: @gunjald I'm not changing the content or meaning of the API.
I'm just applying linting and spelling. These issues you comment need a dedicated PR.

**Comment by nicolacdnll** [comment]
@Kevsy do you know how we can tell the linter how to ignore some files like `code/API_definitions/Discovery/MEC exposure and experience management.yaml`? (I'm asking you just because you added the linter to the repo.)

File: code/API_definitions/Discovery/README.md (hunk 1 above)
Comment (Kevsy) [reply code/API_definitions/Discovery/README.md:5]: > Should it be "platforms"? Plural as the API returns the array of platforms.

No, Simple Edge Discovery returns only the closest platform. For convenience this is returned as an array with one member, to allow easier portability of parsing code to the other Edge Cloud API that will return multiple platforms.

**Review by Kevsy — APPROVED** [review]

**Comment by Kevsy** [comment]
Approved - thanks @nicolacdnll for the good work! - and I will port the new format into #194 (The Simple Edge Discovery PR with substantial content changes). Also noted that the README will need to change in a separate PR.

**Pull request merged**